sycl/include/CL/sycl/ONEAPI/filter_selector.hpp

```diff
@@ -0,0 +1,45 @@
+//==------ filter_selector.hpp - ONEAPI filter selector -------*- C++ --*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/device_selector.hpp>
+
+#include <memory>
+
+namespace std {
+class string;
```

Comment:
I do not think forward declaration is enough because you are taking a string value on line 35.
I think you need to include string header.

Reply:
Good news, because forward declaration of anything in `std::` is forbidden by ISO C++...

Reply:
@keryell is an ISO avenging angel!  No pull request is safe from invalid C++!